### PR TITLE
Next: Module installs integration

### LINKED ISSUE #4591: Create a Nuxt module for integration installation
Currently, the setup of integration happens through `setup`  function putted in `plugins` folder. It would be much better to do this through Nuxt module to follow the most common way of adding 3rd parties in Nuxt. Plugin registration should happen under the hood

# What should be done

1. Inside the integrations, composable package create a `src/nuxt` folder.
2. Inside the folder create `index.js` and `plugin.js`.
3. In `plugin.js` there should be a Nuxt plugin working exactly like `commercetools/theme/plugins/commercetools.js` taking the configuration from Nuxt Module
4. In `index.js` we should export a Nuxt module that accepts options same as options from `setup`. They should be used to register a plugin via `addPlugin` with options passed to module config ( https://nuxtjs.org/api/internals-module-container/#addplugin-template- )

Installation of comemrcetools in `nuxt-config.js` should look more or less like this
```js
[`@vue-storefront/commercetools/nuxt`, {
  // setup config options
}]
```

### PR DESCRIPTION
**Pull request by Fifciu**
Close #4591

### CODE REVIEW

File: packages/commercetools/composables/nuxt/index.js [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+
+export default function (moduleOptions) {
+  this.options.router.middleware.push('commercetools');
+  this.addPlugin({
+    src: path.resolve(__dirname, './plugin.js'),
+    options: moduleOptions
+  });
+
+  const localeNames = moduleOptions.locales.map(l => ({ code: l.name, file: `${l.name}.js`, iso: l.name }));
+  this.options.i18n = {
```
Comment (andrzejewsky) [inline packages/commercetools/composables/nuxt/index.js:11]: that's not related to commercetools, it's an app  setting

File: packages/commercetools/composables/nuxt/plugin.js [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+          'view_published_products:vsf-ct-dev'
+        ]
+      },
+      locale: app.$cookies.get(moduleOptions.cookies.localeCookieName) || moduleOptions.locale,
+      currency: app.$cookies.get(moduleOptions.cookies.currencyCookieName) || moduleOptions.currency,
+      country: app.$cookies.get(moduleOptions.cookies.countryCookieName) || moduleOptions.country
+    }
+  }
+
+Middleware.commercetools = async ({ app }) => {
```
Comment (andrzejewsky) [inline packages/commercetools/composables/nuxt/plugin.js:34]: why middleware is inside of plugin file?

File: packages/commercetools/theme/nuxt.config.js [hunk 3]
```diff
@@ -54,8 +48,57 @@ export default {
...
+          'create_anonymous_token:vsf-ct-dev',
+          'manage_my_orders:vsf-ct-dev',
+          'manage_my_profile:vsf-ct-dev',
+          'manage_my_shopping_lists:vsf-ct-dev',
+          'manage_my_payments:vsf-ct-dev',
+          'view_products:vsf-ct-dev',
+          'view_published_products:vsf-ct-dev'
+        ]
+      },
+      locale: 'en',
```
Comment (andrzejewsky) [inline packages/commercetools/theme/nuxt.config.js:70]: everything besides API should be a default setting inside of the plugin

File: packages/about-you/composables/nuxt/README.MD [hunk 4]
```diff
@@ -0,0 +1,13 @@
+# Example config
+```js
+export const config = {
+  api: {
+    host: 'https://boston.backbone-api.demo.aboutyou.cloud/v1/',
+    auth: {
+      username: 'aboutyou',
+      password: 'OmNErAb96Y5Qn75SFhXr'
```
Comment (andrzejewsky) [inline packages/about-you/composables/nuxt/README.MD:8]: if it's a doc - don't use real access data

File: packages/about-you/theme/helpers/cart/getCartToken.js [hunk 5]
```diff
@@ -1,8 +1,7 @@
-import { generateToken } from '~/helpers/utils/generateToken';
-
-const AYC_CART_TOKEN = 'vsf-ayc-cart-token';
-
 export const getCartToken = context => {
+  // eslint-disable-next-line
+  const { generateToken } = require('@vue-storefront/about-you-theme/helpers/utils/generateToken');
+  const AYC_CART_TOKEN = 'vsf-ayc-cart-token';
```
Comment (andrzejewsky) [inline packages/about-you/theme/helpers/cart/getCartToken.js:4]: lift it up, you don't need to declare that const each time you call function

File: packages/about-you/theme/helpers/wishlist/getWishlistToken.js [hunk 6]
```diff
@@ -1,8 +1,7 @@
-import { generateToken } from '~/helpers/utils/generateToken';
-
-const AYC_WISHLIST_TOKEN = 'vsf-ayc-wishlist-token';
-
 export const getWishlistToken = context => {
+  // eslint-disable-next-line
+  const { generateToken } = require('@vue-storefront/about-you-theme/helpers/utils/generateToken');
+  const AYC_WISHLIST_TOKEN = 'vsf-ayc-wishlist-token';
```
Comment (andrzejewsky) [inline packages/about-you/theme/helpers/wishlist/getWishlistToken.js:4]: same, lift it up

File: packages/about-you/theme/helpers/wishlist/getWishlistToken.js [hunk 7]
```diff
@@ -1,8 +1,7 @@
-import { generateToken } from '~/helpers/utils/generateToken';
-
-const AYC_WISHLIST_TOKEN = 'vsf-ayc-wishlist-token';
-
 export const getWishlistToken = context => {
+  // eslint-disable-next-line
+  const { generateToken } = require('@vue-storefront/about-you-theme/helpers/utils/generateToken');
```
Comment (andrzejewsky) [inline packages/about-you/theme/helpers/wishlist/getWishlistToken.js:3]: why actually do you use require inside of the function, any reason?

File: packages/boilerplate/composables/nuxt/README.MD [hunk 8]
```diff
@@ -0,0 +1,6 @@
+# Example config
+```js
+export const config = {
```
Comment (andrzejewsky) [inline packages/boilerplate/composables/nuxt/README.MD:3]: you don't need nuxt-module for boilerplate

File: packages/commercetools/composables/nuxt/README.MD [hunk 9]
```diff
@@ -0,0 +1,53 @@
+# Example config
+```js
+export const config = {
+  disableMiddleware: false,
+  api: {
+    uri: 'https://api.commercetools.com/vsf-ct-dev/graphql',
```
Comment (andrzejewsky) [inline packages/commercetools/composables/nuxt/README.MD:6]: same - if it's a doc - don't use real access data. Put there something like "your-secret-key" etc.

File: packages/commercetools/composables/nuxt/helpers/index.js [hunk 10]
```diff
@@ -0,0 +1,22 @@
...
+      uri: moduleOptions.api.uri,
+      authHost: moduleOptions.api.authHost,
+      projectKey: moduleOptions.api.projectKey,
+      clientId: moduleOptions.api.clientId,
+      clientSecret: moduleOptions.api.clientSecret,
+      scopes: moduleOptions.api.scopes || defaultConfig.api.scopes
+    },
+    locale: (moduleOptions.cookies && app.$cookies.get(moduleOptions.cookies.localeCookieName)) || moduleOptions.locale || defaultConfig.locale,
+    currency: (moduleOptions.cookies && app.$cookies.get(moduleOptions.cookies.currencyCookieName)) || moduleOptions.currency || defaultConfig.currency,
+    country: (moduleOptions.cookies && app.$cookies.get(moduleOptions.cookies.countryCookieName)) || moduleOptions.country || defaultConfig.country
```
Comment (andrzejewsky) [inline packages/commercetools/composables/nuxt/helpers/index.js:18]: create a following function & lift it up:

```
const getLocaleSettings = (moduleOptions, app) => {
 if (!moduleOptions.cookies) {
   return  {
     locale: moduleOptions.locale || defaultConfig.locale,
     country: moduleOptions.currency || defaultConfig.currenc,
     currency:  moduleOptions.country || defaultConfig.country
   }
 }

   return  {
     locale: $cookies.get(moduleOptions.cookies.localeCookieName),
     country: $cookies.get(moduleOptions.cookies.currencyCookieName),
     currency: $cookies.get(moduleOptions.cookies.countryCookieName)
   }
}
```

Comment (andrzejewsky) [reply packages/commercetools/composables/nuxt/helpers/index.js:18]: and use `... getLocaleSettings(moduleOptions, app)`

File: packages/commercetools/composables/nuxt/index.js [hunk 11]
```diff
@@ -0,0 +1,11 @@
+import path from 'path';
+
+export default function (moduleOptions) {
+  if (!moduleOptions.disableMiddleware && this.options.router && this.options.router.middleware && !this.options.router.middleware.includes('commercetools')) {
```
Comment (andrzejewsky) [inline packages/commercetools/composables/nuxt/index.js:4]: don't do such a long ifs, split them into smaller functions that tell and mean something, for example in this case i can see one that can be split:

```
const hasDefinedMiddleware = (options) => options.router && options.router.middleware && options.router.middleware.includes('commercetools')
```

then 
```
if (!moduleOptions.disableMiddleware && !hasDefinedMiddleware(this.options)) 
```
cleaner?

File: packages/commercetools/composables/nuxt/index.js [hunk 12]
```diff
@@ -0,0 +1,11 @@
...
+    this.options.router.middleware.push('commercetools');
```
[the remark this answers is not in the document]
Comment (Fifciu) [reply packages/commercetools/composables/nuxt/index.js:8]: Just added this to the config `export const CT_TOKEN_MIDDLEWARE_SLUG = 'commercetools-generate-access-token';` and importing it when needed

File: packages/commercetools/composables/nuxt/plugin.js [hunk 13]
```diff
@@ -0,0 +1,45 @@
+/* eslint-disable */
+import { setup } from '@vue-storefront/commercetools-api';
+import { mapConfigToSetupObject, CT_TOKEN_COOKIE_NAME } from '@vue-storefront/commercetools/nuxt/helpers'
+
+const moduleOptions = JSON.parse('<%= JSON.stringify(options) %>');
+
+<% if (!options.disableMiddleware) { %>
```
[the remark this answers is not in the document]
Comment (Fifciu) [reply packages/commercetools/composables/nuxt/plugin.js:7]: Changed to `disableGenerateTokenMiddleware`

File: packages/boilerplate/composables/nuxt/plugin.js [hunk 14]
```diff
@@ -0,0 +1,7 @@
+import { setup } from '@vue-storefront/boilerplate-api';
+
+export default function init() {
+  // const moduleOptions = JSON.parse('<%= JSON.stringify(options) %>');
```
Comment (filrak) [inline packages/boilerplate/composables/nuxt/plugin.js:4]: remove

**Pull request merged**